Trakt/ScheduledTasks/SyncLibraryTask.cs

```diff
@@ -127,81 +127,94 @@ public async Task ExecuteAsync(IProgress<double> progress, CancellationToken can
-                        OrderBy = new[]
-                        {
-                            (ItemSortBy.SortName, SortOrder.Ascending)
-                        }
-                    })
-                .Where(x => _traktApi.CanSync(x, traktUser))
-                .ToList();
+
+        var baseQuery = new InternalItemsQuery(user)
+        {
+            IncludeItemTypes = new[] { BaseItemKind.Movie },
+            IsVirtualItem = false,
+            OrderBy = new[]
+            {
+                (ItemSortBy.SortName, SortOrder.Ascending)
+            }
+        };
+
+        var totalCount = _libraryManager.GetCount(baseQuery);
+        var decisionProgress = progress.Split(4).Split(totalCount);
+
         var collectedMovies = new List<Movie>();
         var playedMovies = new List<Movie>();
         var unplayedMovies = new List<Movie>();
 
-        var decisionProgress = progress.Split(4).Split(libraryMovies.Count);
-        foreach (var child in libraryMovies)
+        const int Limit = 100;
+        int offset = 0, previousCount;
+
+        do
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            var libraryMovie = child as Movie;
-            var userData = _userDataManager.GetUserData(user.Id, child);
+            baseQuery.Limit = Limit;
+            baseQuery.StartIndex = offset;
 
-            if (traktUser.SynchronizeCollections)
+            var libraryMovies = _libraryManager.GetItemList(baseQuery);
+            previousCount = libraryMovies.Count;
+            offset += Limit;
+
+            foreach (var libraryMovie in libraryMovies.OfType<Movie>().Where(x => _traktApi.CanSync(x, traktUser)))
             {
-                // if movie is not collected, or (export media info setting is enabled and every collected matching movie has different metadata), collect it
-                var collectedMathingMovies = SyncFromTraktTask.FindMatches(libraryMovie, traktCollectedMovies).ToList();
-                if (!collectedMathingMovies.Any()
-                    || (traktUser.ExportMediaInfo
-                        && collectedMathingMovies.All(
-                            collectedMovie => collectedMovie.MetadataIsDifferent(libraryMovie))))
+                cancellationToken.ThrowIfCancellationRequested();
+                var userData = _userDataManager.GetUserData(user.Id, libraryMovie);
+
+                if (traktUser.SynchronizeCollections)
                 {
-                    collectedMovies.Add(libraryMovie);
+                    // if movie is not collected, or (export media info setting is enabled and every collected matching movie has different metadata), collect it
+                    var collectedMathingMovies = SyncFromTraktTask.FindMatches(libraryMovie, traktCollectedMovies).ToList();
+                    if (!collectedMathingMovies.Any()
+                        || (traktUser.ExportMediaInfo
+                            && collectedMathingMovies.All(
```

Comment:
I guess this should be `collectedMatchingMovies`?

Reply:
Should i make this change in my fork as well?

Reply:
Likely, but I didn't change that :)

Reply:
It's a local variable, so not needed. I'll fix the typo after I finish testing